### ISSUE TITLE
Задание не выполняется по расписанию.

### ISSUE THREAD

**Issue by uramalyutin** [ivanov660/TestingTool-3#18]

**Comment by ivanov660** [comment]
Добрый день!

1. Посмотрите логи в журнале регистрации, система обычно туда пишет о всех возможных проблемах.
2. Скорее всего проблема в том, что для пользователя под которым выполняются задания не настроены каталоги по умолчанию. Об этом информирует статус "отменено".
Проверьте следующие основные моменты:
а) Администрирование -> Консоль заданий -> Для регламентных заданий "Обработка заданий" и "Обработка действий заданий" установлен пользователь , к примеру, "Администратор" (с полными правами). Пустой не верно.
б) Администрирование -> Настройки работы пользователя на рабочем месте -> Для пользователя "Администратор" установлены настройки на рабочем месте (имя сервера) следующие параметры: "Путь к исполняемому файлу 1С", "Путь к каталогу наборов сценариев", "Путь к каталогу отчетов выполнения тестов", "Путь к каталогу библиотеки сценариев". Это обязательные настройки для выполнения сценарных тестов. Каталоги должны быть доступны службе под которой запускается служба 1С.

Еще не забудьте в настройках разрешить интерактивное взаимодействие с рабочим столом.

**Comment by ivanov660** [comment]
Если данное утверждение подтвердиться, тогда думаю, что необходимо сделать информер о проблемах в настройке выполнения заданий и высвечивать для администраторов.

**Comment by uramalyutin** [comment]
Владимир, спасибо за ответ, он помог.
Проблема была в том, что у меня вообще не были добавлены регламентные задания (мог бы и догадаться, что кто-то же должен запускать выполнение заданий =), т.к. базу я создавал чистую из *.cf и настраивал все полностью с нуля.

Ххотя я и сдвинулся с места и мой пробный тест теперь нормально запускается, но, к сожалению, не выполняется: не создается документ в тестируемой базе. В каталог отчетов падает файл, но, видимо, не загружается в базу, т.к. не видно информации о результатах выполнения тестов, хотя файл из каталога через некоторое время файл пропадает. Я успел сохранить, его и сравнить с отчетом, полученным после ручного запуска из обработки "МенеджерСценарногоТеста.epf" (при ручном выполнении, повторюсь, никаких проблем или ошибок нет), кроме времени выполнения есть несколько различий: в начале файла один из шагов не выполнился, однако выполнение теста продолжилось и немного далее еще один шаг не выполнился, все последующие шаги при этом были отменены. 

Мучаюсь уже почти целый день, настройки перепроверил несколько раз, пока не добился успеха. Снова надеюсь на вашу помощь :)


![01](https://user-images.githubusercontent.com/11212311/46472997-8b1b6000-c7e7-11e8-96bd-72c2368e6f63.png)
![02](https://user-images.githubusercontent.com/11212311/46472999-8b1b6000-c7e7-11e8-84d8-2c78f59fd9fd.png)
![03](https://user-images.githubusercontent.com/11212311/46473000-8bb3f680-c7e7-11e8-895a-0fbdc8ed5d1e.png)
![04](https://user-images.githubusercontent.com/11212311/46473001-8bb3f680-c7e7-11e8-90b0-9380c6c51ea0.png)
![05](https://user-images.githubusercontent.com/11212311/46473002-8bb3f680-c7e7-11e8-8d40-708a39d14674.png)

**Comment by ivanov660** [comment]
1. В текущий момент система возвращает сообщения о результатах взаимодействия через Automation API от 1С. В процедуре подключения выполняется поиск окна, и если оно не найдено, тогда выдается ошибка. Получить более подробную информацию проблематично средствами 1С. 

Как понять проблему:
1. Можно открыть нулевой сеанс во время выполнения теста (запуск процедуры Interactive Services Detection) или залогиниться под пользователем и посмотреть цикл выполнения теста. Увидеть ошибку и станет ясно. **Это пригодится в любом случае**
2. Окно может не находится, в начале сценария. Когда запускается 1С и висят служебные окна перед запуском (не обнаружена лицензия и т.п.)
3. В процессе выполнения сценария окна могут не появляться в зависимости от настроек пользователя в программе.  К примеру, у него отключено окно запроса оборотной тары. Т.е. при записи сценария у вас окно есть, а у него нет.
Судя по скриншоту, у вас как раз именно эта ситуация - служебное окно "1С Предприятие" и в случае тестирующего пользователя оно по каким-то причинам не появляется.
Для этих целей, я рекомендую использовать команду "**Элемент существует?**" Как это работает:
а) Вы добавляете этот шаг и прописываете заголовок объекта "1С Предприятие....правильно"
б) Добавляете переменную в параметры, к примеру, ОкноСуществует, и указываете в поле свойтсво
в) После добавляете ребенка "Условие" и в нем сравниваете ОкноСуществует (установите Имя Параметра "ОкноСуществует", ЗначениеПараметра "Да", Условие "Равно")
г) После условия повесьте цепочку обработки этого окна.

Таким образом, если окно/элемент будет найден, то скрипт зайдет внутрь иначе пропустит эту цепочку.

2. Сейчас мы дорабатываем менеджер для поддержки других API и в рамках данной процедуры попробуем подключить формирование скриншота (что-бы зафиксировать наглядную картинку). Allure формат как раз поддерживает скриншоты.

**Comment by uramalyutin** [comment]
Владимир, подскажите еще как открыть этот нулевой сеанс?
mstsc /v servername /console - запускается, но окно 1С в этом сеансе не видно, видно только процесс в диспетчере задач. Служба Interactive Services Detection запущена. Ни на каком этапе ошибки не было, в результате я захожу на сервер под пользователем, под которым запущен сервер 1С, но окно клиента тестирования не видно.

**Comment by ivanov660** [comment]
На сколько я понял, то у вас сервер версии 2012 и выше. (к примеру, XP запускается в текущем рабочем столе)
1. Необходимо убедиться, что агент 1С (ragent.exe) в Службах (Services) запущен с настройкой  "Allow service to interact with desktop" (Взаимодействие с рабочим столом). В этом случае используется всегда "Local System account".
API Autoamtion от 1C при отключенной опции иногда некорректно работает (у меня всегда была ошибка при команде "Перейти к строке").
2. Убедиться что запущена служба "Interactive Services Detection"
3. У пользователя под которым вы логинитесь есть право взаимодействия (лучше всего администратор)
4. По соблюдении этих правил у вас на рабочем столе появится окошко с заголовком "Interactive Services Detection". Чтобы открыть рабочий стол нулевого сеанса нужно нажать кнопку "View the message".

![image](https://user-images.githubusercontent.com/24763549/46718640-80901900-cc74-11e8-89f5-897f69d9c70f.png)

**Comment by ivanov660** [comment]
@uramalyutin удалось разобраться?

**Comment by uramalyutin** [comment]
Добрый день.
Пока не удалось. В силу разных причин мы не можем запустить 1С от системной учетной записи. Поэтому приняли решение создать отдельный сервер, на котором 1С будет работать под системной учеткой, на нем развернуть копию базы, на него же перенести TestingTool и уже разбираться в чем дело. В итоге я пока жду когда будет сервер и можно будет продолжить работы.
Не закрывайте пока этот запрос, я обязательно в нем отпишусь о результатах.

**Comment by ivanov660** [comment]
Ок. Ждем обратную связь по результатам изменения организационной структуры тестового стенда.

**Comment by uramalyutin** [comment]
Добрый день.
Мне наконец удалось сделать все так, как вы рекомендовали. И после этого произошло одно интересное событие: тест выполнился от начала и до конца без ошибок (как и при запуске из обработки "Менеджер сценарного теста", только когда я стал наблюдать за выполнением. Следующий запуск уже без наблюдения - выполнился без ошибок. Хотя два запуска проверки, которые были выполнены до того, как я стал наблюдать за выполнением, "упали" на 67 шаге.
Правда между "упавшими" проверками и выполнившимися были две проверки, которые я отменил сам (не думаю, что это может как-то повлиять на результат).
Иначе как теорией квантовой неопределенности я это объяснить не могу :)
Но в любом случае спасибо вам за помощь, буду изучать дальше и пробовать создавать новые тесты.
![image](https://user-images.githubusercontent.com/11212311/47485387-b7565980-d846-11e8-8e74-533953654a75.png)

**Comment by ivanov660** [comment]
Обсуждение закрываем?

**Comment by uramalyutin** [comment]
Да, конечно. Заявленная изначально проблема решена. Если будут еще проблемы, я сделаю новый запрос. Наверное не стоит разные проблемы в одну кучу валить.
